### PR TITLE
Added methods to enhance datetime class.

### PR DESCRIPTION
**Pull request by kaviranjan88**
- New Methods for datetime class
- Added unit test

### CODE REVIEW

**Comment by gavv** [comment]
Hi, thanks for pr!

A couple of comments.

**Comment by gavv** [comment]
For each method, could you please add a short comment describing the return value?

For example, for Month we should mention the range of returned integer (is it 0-11 or 1-12).

You can base comments on comments from time.Time.

**Comment by gavv** [comment]
And another one. 

For the test, could you use a date-time which fields are all nonzero and all have different numeric values?

This is needed to be sure that the methods work properly.

**Comment by gavv** [comment]
Let me know when it's ready for review.

**Comment by kaviranjan88** [comment]
@gavv Hi, I have updated code as per your comments. Please have a look.

File: datetime.go [hunk 1]
```diff
@@ -283,3 +283,214 @@ func (dt *DateTime) NotInRange(min, max time.Time) *DateTime {
...
+	defer dt.chain.leave()
+
+	if dt.chain.failed() {
+		return newNumber(dt.chain, float64(0))
+	}
+
+	return newNumber(dt.chain, float64(dt.value.Year()))
+}
+
+// Month returns the month of the year specified by datetime.
```
Comment (gavv) [inline datetime.go:322]: Let's also specify the range of returned value.

File: datetime.go [hunk 2]
```diff
@@ -283,3 +283,214 @@ func (dt *DateTime) NotInRange(min, max time.Time) *DateTime {
...
+	defer dt.chain.leave()
+
+	if dt.chain.failed() {
+		return newNumber(dt.chain, float64(0))
+	}
+
+	return newNumber(dt.chain, float64(dt.value.Month()))
+}
+
+// Day returns the day of the month specified datetime.
```
Comment (gavv) [inline datetime.go:339]: Let's also specify the range of returned value.

File: datetime.go [hunk 3]
```diff
@@ -283,3 +283,214 @@ func (dt *DateTime) NotInRange(min, max time.Time) *DateTime {
...
+	defer dt.chain.leave()
+
+	if dt.chain.failed() {
+		return newNumber(dt.chain, float64(0))
+	}
+
+	return newNumber(dt.chain, float64(dt.value.Day()))
+}
+
+// Weekday returns the day of the week specified by datetime.
```
Comment (gavv) [inline datetime.go:356]: Let's also specify the range of returned value.

File: datetime.go [hunk 4]
```diff
@@ -283,3 +283,214 @@ func (dt *DateTime) NotInRange(min, max time.Time) *DateTime {
...
+
+	if dt.chain.failed() {
+		return newNumber(dt.chain, float64(0))
+	}
+
+	return newNumber(dt.chain, float64(dt.value.YearDay()))
+}
+
+// Hour returns the hour within the day specified by datetime,
+// in the range [0, 23]..
```
Comment (gavv) [inline datetime.go:393]: Extra dot

File: datetime.go [hunk 5]
```diff
@@ -283,3 +283,214 @@ func (dt *DateTime) NotInRange(min, max time.Time) *DateTime {
...
+
+	if dt.chain.failed() {
+		return newNumber(dt.chain, float64(0))
+	}
+
+	return newNumber(dt.chain, float64(dt.value.Hour()))
+}
+
+// Minute returns the minute offset within the hour specified by datetime,
+// in the range [0, 59]..
```
Comment (gavv) [inline datetime.go:411]: Extra dot

**Review by gavv — COMMENTED** [review]
Thanks for update. A few more comments

**Comment by gavv** [comment]
Regarding code examples in comments:

* instead of unix(0,0), I suggest to use time.Parse with a date with distinct fileds, to make examples more convenient 
* when calling Equal, cast to float64 is not needed

File: datetime.go (hunk 1 above)
Comment (kaviranjan88) [reply datetime.go]: Updated

File: datetime.go (hunk 2 above)
Comment (kaviranjan88) [reply datetime.go]: Done

File: datetime.go (hunk 3 above)
Comment (kaviranjan88) [reply datetime.go]: Done

File: datetime.go (hunk 4 above)
Comment (kaviranjan88) [reply datetime.go]: Done

File: datetime.go (hunk 5 above)
Comment (kaviranjan88) [reply datetime.go]: Done

**Comment by kaviranjan88** [comment]
@gavv Hi, I have updated code as per your comments. Casting to float64 as time returns **int** while number class has return type as **float64**. Please suggest in case I am not doing it correct.

**Comment by gavv** [comment]
Sorry, I was talking about code on examples in documentation comments - I suggested to use time.Parse and casts there, not in the test.

Regarding casts. Yes, they are needed in test. But seems they are not needed in those code examples because they use httpexpect's Equal, which allow numbers of different types (fliat and int).

**Comment by kaviranjan88** [comment]
@gavv Hi, Updated examples as per your comments. Please have a look.

File: datetime.go [hunk 6]
```diff
@@ -283,3 +283,230 @@ func (dt *DateTime) NotInRange(min, max time.Time) *DateTime {
 
 	return dt
 }
+
+// Zone returns a new String instance with datetime zone.
+//
+// Example:
+//
+//	tm, _ := time.Parse(time.UnixDate, "FRI Dec 30 15:04:05 IST 2022")
```
Comment (gavv) [inline datetime.go:291]: Could we replace "FRI" with "Fri" in examples? Usually it's how days are spelled.

File: datetime.go [hunk 7]
```diff
@@ -283,3 +283,230 @@ func (dt *DateTime) NotInRange(min, max time.Time) *DateTime {
...
+
+	if dt.chain.failed() {
+		return newNumber(dt.chain, float64(0))
+	}
+
+	return newNumber(dt.chain, float64(dt.value.Day()))
+}
+
+// Weekday returns the day of the week specified by datetime,
+// in the range [0, 6].
```
Comment (gavv) [inline datetime.go:364]: We should mention in comment that 0 corresponds to Sunday.

File: datetime.go [hunk 8]
```diff
@@ -283,3 +283,230 @@ func (dt *DateTime) NotInRange(min, max time.Time) *DateTime {
...
+//
+// Example:
+//
+//	tm, _ := time.Parse(time.UnixDate, "FRI Dec 30 15:04:05 IST 2022")
+//	dt := NewDateTime(t, tm)
+//	dt.WeekDay().Equal(5)
```
Comment (gavv) [inline datetime.go:370]: Instead of 5, let's use time.Friday constant

**Comment by gavv** [comment]
Thanks, LGTM!

**Pull request merged**